clean_all.py

```diff
@@ -0,0 +1,265 @@
+            self._refresh_locations_and_zones(force=True)
+        return self._zones
+
+    def _delete_all(self, endpoint_name: str) -> None:
+        """
+        Iterates through all zones, lists object under given ``endpoint_name``
+        and then deletes those objects.
+        """
+        for zone in self.zones:
+            logger.info(f"Deleting compute {endpoint_name} in {zone}")
+            endpoint = getattr(self.client, endpoint_name)()
+            for obj in self._iterate(
+                endpoint=endpoint, project=self.project_id, zone=zone
+            ):
+                if SKIP_LABEL not in obj.get("labels", {}) and self.is_stale(
+                    obj["creationTimestamp"]
+                ):
+                    self._delete(
+                        name=endpoint_name,
+                        endpoint=endpoint,
+                        instance=obj,
+                        project=self.project_id,
+                        zone=obj["zone"].split("/")[-1],
+                        obj=obj["id"],
+                    )
+
+    def delete_all_disks(self) -> None:
+        self._delete_all("disks")
+
+    def delete_all_instances(self) -> None:
+        self._delete_all("instances")
+
+
+class DataprocClient(BaseDiscoveryClient):
+    endpoint = "dataproc"
+
+    def _delete_all_in_location(self, location: str):
+        clusters = self._iterate(
+            endpoint=self.client.projects().regions().clusters(),
+            key="clusters",
+            projectId=self.project_id,
+            region=location,
+        )
+        for cluster in clusters:
+            last_state_date = cluster["status"]["stateStartTime"]
+            if SKIP_LABEL not in cluster.get("labels", []) and self.is_stale(
+                last_state_date
+            ):
+                self._delete(
+                    name="cluster",
+                    endpoint=self.client.projects().regions().clusters(),
+                    instance=cluster,
+                    key="clusterName",
+                    projectId=self.project_id,
+                    region=location,
+                    clusterName=cluster["clusterName"],
+                )
+
+    def delete_all_clusters(self, locations: List[str]):
+        self._delete_in_all_locations(
+            locations=locations, object_name="dataproc clusters"
+        )
+
+
+class ComposerClient(BaseDiscoveryClient):
+    endpoint = "composer"
+
+    def _delete_all_in_location(self, location: str) -> None:
+        conn = self.client.projects().locations().environments()
+        environments = self._iterate(
+            endpoint=conn,
+            key="environments",
+            parent=f"projects/{self.project_id}/locations/{location}",
+        )
+        for env in environments:
+            if SKIP_LABEL not in env.get("labels", {}) and self.is_stale(
+                env["updateTime"]
+            ):
+                self._delete(name="composer", endpoint=conn, key="name", instance=env)
```

Comment:
We need to pass the environment name in format "projects/{project_id}/locations/{location}/environments/{environment}" to the delete request. It should be available in `env` object under `name` key. However, since the argument used in request is also called `name`, then I guess the same argument of `_delete` method must be renamed.